### PR TITLE
Counting and pagination by dense rank

### PR DESCRIPTION
**Pull request by zerenato**
Inclusão de novos métodos

- SelectCount e SelectCountAll
- PrependSelect: Adiciona campos ao select
- PaginatedByDenseRank: Faz paginação utilizando o DENSE_RANK

### CODE REVIEW

**Comment by paulohenriquesi** [comment]
Hi, @zerenato!

I understand the reason why you created the prepend method, but I don't think that it needs to be public. It's usage makes the API much less fluent than what we desire. What about make it internal?

**Review by paulohenriquesi — CHANGES_REQUESTED** [review]
Hi, @zerenato!

I understand the reason why you created the prepend method, but I don't think that it needs to be public. It's usage makes the API much less fluent than what we desire. What about make it internal?

File: src/Braspag.FluentQueryBuilder/SelectBuilder.cs [hunk 1]
```diff
@@ -156,6 +195,28 @@ public SelectBuilder Paginated(int pageSize, int currentPage)
             return this;
         }
 
+        /// <summary>
+        /// Adds pagination using DENSE_RANK
+        /// </summary>
+        /// <param name="pageSize">Number of rows per page</param>
+        /// <param name="currentPage">Current page</param>
+        /// <param name="rankField">Field used for ranking</param>
+        public WhereBuilder PaginatedByDenseRank(int pageSize, int currentPage, string rankField)
```
Comment (tiagorosendo) [inline src/Braspag.FluentQueryBuilder/SelectBuilder.cs]: I thik that will be better if you change the method "Paginated" and add an option to select the paginated type (OffSet or DenseRank). This method may continue to exist as a private method for use in Paginated method

**Review by tiagorosendo — CHANGES_REQUESTED** [review]

File: src/Braspag.FluentQueryBuilder/SelectBuilder.cs (hunk 1 above)
Comment (zerenato) [reply src/Braspag.FluentQueryBuilder/SelectBuilder.cs]: Those two methods return different types.
"Paginated" returns SelectBuilder and "PaginatedByDenseRank" returns a WhereBuilder.
I can create a "ToSelectBuilder" method on WhereBuilder and make the suggested changes or leave it as it is.

Comment (zerenato) [reply src/Braspag.FluentQueryBuilder/SelectBuilder.cs]: Done

**Review by tiagorosendo — DISMISSED** [review]

**Review by tiagorosendo — DISMISSED** [review]

**Review by paulohenriquesi — DISMISSED** [review]

**Review by paulohenriquesi — APPROVED** [review]

**Pull request merged**